### ISSUE TITLE
parrot_run leaves a remaining process if it doesn't have the correct caps

### ISSUE THREAD

**Issue by PerilousApricot** [cooperative-computing-lab/cctools#2284]
Hello,

While trying to bolt chirp+parrot into spark on the CERN SWAN service, the process dies because the container they launch the user in doesn't have the right caps, which is OK. I then noticed, however that there is a process leftover which spins @ 100% cpu indefinitely:

```
bash-4.2$ ./parrot_run ls
The `ptrace` system call appears to be disabled.
Some possible causes:
 - Syscall filtering (e.g. seccomp) is in place. Some versions of Docker do
   this inside containers.
 - The program that launched Parrot used `PR_SET_DUMPABLE` to disable debugging
   for this process.
 - Your system's security framework (SELinux, Yama, etc.) disables ptrace.
2020/04/23 18:39:53.16 parrot_run[1495] fatal: could not trace child
2020/04/23 18:39:53.16 parrot_run[1495] notice: received signal 15 (Terminated), killing all my children...
2020/04/23 18:39:53.16 parrot_run[1495] notice: sending myself 15 (Terminated), goodbye!
Terminated
bash-4.2$ echo $$
466
bash-4.2$ ps aux | grep 466
meloam     466  0.0  0.0  13564  2204 pts/0    Ss   18:16   0:01 /bin/bash -l
meloam    1498  0.0  0.0   8428   888 pts/0    S+   18:40   0:00 grep --color 466
bash-4.2$ ps aux | grep parrot
meloam    1496  101  0.0  70100  1468 pts/0    R    18:39   0:20 ./parrot_run ls
```

This is with cctools-7.1.2-x86_64-centos7. Here's info about the environment if that helps:
```
bash-4.2$ uname -a
Linux 7c1961358e29 3.10.0-1062.18.1.el7.x86_64 #1 SMP Tue Mar 17 23:49:17 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
bash-4.2$ capsh --print
Current: = cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_raw,cap_mknod,cap_audit_write,cap_setfcap+i
Bounding set =cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_raw,cap_mknod,cap_audit_write,cap_setfcap
Securebits: 00/0x0/1'b0
 secure-noroot: no (unlocked)
 secure-no-suid-fixup: no (unlocked)
 secure-keep-caps: no (unlocked)
uid=50259(meloam)
gid=50259(meloam)
groups=50259(meloam
```

**Comment by PerilousApricot** [comment]
A bit more context:
```
The `ptrace` system call appears to be disabled.
Some possible causes:
 - Syscall filtering (e.g. seccomp) is in place. Some versions of Docker do
   this inside containers.
 - The program that launched Parrot used `PR_SET_DUMPABLE` to disable debugging
   for this process.
 - Your system's security framework (SELinux, Yama, etc.) disables ptrace.
2020/04/23 19:13:07.46 parrot_run[1738] fatal: could not trace child
2020/04/23 19:13:07.46 parrot_run[1738] notice: received signal 15 (Terminated), killing all my children...
2020/04/23 19:13:07.46 parrot_run[1738] notice: sending myself 15 (Terminated), goodbye!
Terminated
bash-4.2$ killall parrot_run
2020/04/23 19:13:11.69 parrot_run[1739] notice: received signal 15 (Terminated), killing all my children...
2020/04/23 19:13:11.69 parrot_run[1739] notice: sending myself 15 (Terminated), goodbye!
```
I presume that since they have sequential PIDs, this a fork child/parent

**Comment by btovar** [comment]
@PerilousApricot  thanks! 

I have been able to reproduce your report.  A parrot_run process does seem to keep alive after the main process terminates.

**Comment by btovar** [comment]
Fixed with #2285
